src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.NamedRequest.cs

```diff
@@ -903,6 +903,44 @@ public async Task accessing_non_existent_header_names_produces_an_error(string p
                 diagnostics.Diagnostics.First().Message.Should().Be($$$"""The supplied header name '{{{headerName}}}' does not exist in the named request.""");
             }
 
+            [Fact]
+            public async Task accessing_an_index_in_a_json_array_succeeds()
+            {
+                using var kernel = new HttpKernel();
+                var firstRequest = """                    
+                    # @name sampleArray
+                    POST https://httpbin.org/anything
+                    {
+                      "devices": [
+                        {
+                          "id": "5601db0f-32e0-4d82-bc79-251e50fa1407",
+                          "name": "Foo"
+                        },
+                        {
+                          "id": "455301a5-8a6e-49d0-b056-96fb2847be18",
+                          "name": "Bar"
+                        }
+                      ]
+                    }
+                    ###
+                    """;
+                var firstResult = await kernel.SendAsync(new SubmitCode(firstRequest));
+                firstResult.Events.Should().NotContainErrors();
+                var secondRequest = """
+                    GET https://httpbin.org/headers
+                    X-Value: {{sampleArray.response.body.$.json.devices[0].id}}
```

Comment:
It might be worthwhile to test a case where there's more than one array index in the expression.